### ISSUE TITLE
Stremio closes 10 seconds after launch (Fedora 38, flatpak, KDE)

### ISSUE THREAD

**Issue by urisimchoni** [Stremio/stremio-bugs#463]
**Describe the bug**
I've been a happy user of Stremio for some time now. Today, without any system updates that I can recall, it started "crashing" shortly after launching the application. By "crashing" I mean that the application just exits without any apparent reason, but no crash event is registered at the system level.

The problem persists after applying all system updates, after reinstalling Stremio (via Flatpak) and also after deleting all state (deleting ~/.var/app/com.stremio.Stremio).

Is there some logging I can enable or a log file I can look at, to get some clue as to what is hapening?

Thanks,
Uri

**Desktop (please complete the following information):**
 - OS: Fedora 38 KDE Spin (Stremio installed via Flatpak)
 - Browser Google-Chrome
 - Version 114.0.5735.90

**Comment by TRtomasz** [comment]
run it from terminal and you will be able to see logs then.

**Comment by urisimchoni** [comment]
This is what I get: (those are the init messages, after 10 seconds the app just exits without any further messages)

[uri@uri-dell applications]$ /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=/app/opt/stremio/stremio --file-forwarding com.stremio.Stremio
[cplayer] mpv 0.33.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
[cplayer]  built on UNKNOWN
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.31.100
[cplayer]    libavcodec      58.54.100
[cplayer]    libavformat     58.29.100
[cplayer]    libswscale      5.5.100
[cplayer]    libavfilter     7.57.100
[cplayer]    libswresample   3.5.100
[cplayer] FFmpeg version: 4.2.2
[cplayer] 
[cplayer] Configuration: waf configure --prefix=/app --enable-libmpv-shared --disable-cplayer --disable-build-date --disable-alsa
[cplayer] List of enabled features: asm cplugins cuda-hwaccel cuda-interop debug-build drm egl egl-drm egl-helpers egl-x11 ffmpeg ffnvcodec gbm gbm.h gl gl-wayland glibc-thread-name glob glob-posix gpl iconv jpeg lcms2 libarchive libass libavdevice libdl libm libmpv-shared librt linux-fstatfs linux-input-event-codes memfd_create optimize plain-gl posix posix-or-mingw pthreads pulse stdatomic uchardet vaapi vaapi-drm vaapi-egl vaapi-wayland vaapi-x-egl vaapi-x11 vdpau vt.h wayland wayland-protocols x11 xv zlib
[cplayer] Set property: vo="libmpv" -> 1
[cplayer] Set property: gpu-hwdec-interop="auto" -> 1
[cplayer] Set property: cache-default=15000 -> -3
[cplayer] Set property: cache-backbuffer=15000 -> -3
Warning: option --cache-secs is deprecated and might be removed in the future (will use unlimited time).
[cplayer] Set property: cache-secs=10 -> 1
[cplayer] Set property: audio-client-name="Stremio" -> 1
[cplayer] Set property: title="Stremio" -> 1
[cplayer] Set property: audio-fallback-to-null="yes" -> 1
[cplayer] Set property: audio-client-name="Stremio" -> 1
[cplayer] Set property: title="Stremio" -> 1
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
qml: Loading web UI from URL: https://app.strem.io/shell-v4.4/#?loginFlow=desktop
qml: Stremio Shell version: 4.4.160
qml:  **** Completed. Loading Autoupdater ***
qml: Auto-updater: skipping, possibly not running an installed app?
[libmpv_render] GL_VERSION='4.6 (Compatibility Profile) Mesa 21.3.9 (git-78c96ae5b6)'
[libmpv_render] Detected desktop OpenGL 4.6.
[libmpv_render] GL_VENDOR='Intel'
[libmpv_render] GL_RENDERER='Mesa Intel(R) UHD Graphics 620 (WHL GT2)'
[libmpv_render] GL_SHADING_LANGUAGE_VERSION='4.60'
[libmpv_render] GL_*_swap_control extension missing.
[libmpv_render] Testing FBO format rgba16f
[libmpv_render] Using FBO format rgba16f.
[libmpv_render] No advanced processing required. Enabling dumb mode.
[libmpv_render] Loading hwdec driver 'vaapi-egl'
[libmpv_render/vaapi-egl] VAAPI hwdec only works with OpenGL or Vulkan backends.
[libmpv_render] Loading failed.
[libmpv_render] Loading hwdec driver 'cuda-nvdec'
Cannot load libcuda.so.1
[libmpv_render/cuda-nvdec] Failed to load CUDA symbols
[libmpv_render] Loading failed.
[libmpv_render] Loading hwdec driver 'drmprime-drm'
[libmpv_render/drmprime-drm] Failed to retrieve DRM fd from native display.
[libmpv_render] Loading failed.
hls executables located ->  {
  ffmpeg: '/usr/bin/ffmpeg',
  ffsplit: null,
  ffprobe: '/usr/bin/ffprobe'
}
Using app path -> /home/uri/.stremio-server
Enabling casting...
(node:37) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
EngineFS server started at http://127.0.0.1:11470
EngineFS HTTPS endpoint at https://local.strem.io:12470
-> GET /local-addon/manifest.json 
-> GET /settings 
Discovery of new tv device "e2130f32-4749-d77d-9277-56ccc63c5844" - [LG] webOS TV OLED55C1PVA

**Comment by IceWreck** [comment]
I've encountered this before on fedora, run flatpak update and try updating base flatpak environments

**Comment by vesavlad** [comment]
same story here
```
➜  app flatpak run com.stremio.Stremio 
[cplayer] mpv 0.33.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
[cplayer]  built on UNKNOWN
[cplayer] FFmpeg library versions:
[cplayer]    libavutil       56.31.100
[cplayer]    libavcodec      58.54.100
[cplayer]    libavformat     58.29.100
[cplayer]    libswscale      5.5.100
[cplayer]    libavfilter     7.57.100
[cplayer]    libswresample   3.5.100
[cplayer] FFmpeg version: 4.2.2
[cplayer] 
[cplayer] Configuration: waf configure --prefix=/app --enable-libmpv-shared --disable-cplayer --disable-build-date --disable-alsa
[cplayer] List of enabled features: asm cplugins cuda-hwaccel cuda-interop debug-build drm egl egl-drm egl-helpers egl-x11 ffmpeg ffnvcodec gbm gbm.h gl gl-wayland glibc-thread-name glob glob-posix gpl iconv jack jpeg lcms2 libarchive libass libavdevice libdl libm libmpv-shared librt linux-fstatfs linux-input-event-codes memfd_create optimize plain-gl posix posix-or-mingw pthreads pulse stdatomic uchardet vaapi vaapi-drm vaapi-egl vaapi-wayland vaapi-x-egl vaapi-x11 vdpau vt.h wayland wayland-protocols x11 xv zlib
[cplayer] Set property: vo="libmpv" -> 1
[cplayer] Set property: gpu-hwdec-interop="auto" -> 1
[cplayer] Set property: cache-default=15000 -> -3
[cplayer] Set property: cache-backbuffer=15000 -> -3
Warning: option --cache-secs is deprecated and might be removed in the future (will use unlimited time).
[cplayer] Set property: cache-secs=10 -> 1
[cplayer] Set property: audio-client-name="Stremio" -> 1
[cplayer] Set property: title="Stremio" -> 1
[cplayer] Set property: audio-fallback-to-null="yes" -> 1
[cplayer] Set property: audio-client-name="Stremio" -> 1
[cplayer] Set property: title="Stremio" -> 1
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
qml: Loading web UI from URL: https://app.strem.io/shell-v4.4/#?loginFlow=desktop
qml: Stremio Shell version: 4.4.165
qml:  **** Completed. Loading Autoupdater ***
qml: Auto-updater: skipping, possibly not running an installed app?
[libmpv_render] GL_VERSION='4.6 (Compatibility Profile) Mesa 23.3.0 (git-1fbdd37d4c)'
[libmpv_render] Detected desktop OpenGL 4.6.
[libmpv_render] GL_VENDOR='Intel'
[libmpv_render] GL_RENDERER='Mesa Intel(R) Graphics (ADL GT2)'
[libmpv_render] GL_SHADING_LANGUAGE_VERSION='4.60'
[libmpv_render] GL_*_swap_control extension missing.
[libmpv_render] Testing FBO format rgba16f
[libmpv_render] Using FBO format rgba16f.
[libmpv_render] No advanced processing required. Enabling dumb mode.
[libmpv_render] Loading hwdec driver 'vaapi-egl'
[libmpv_render/vaapi-egl] using VAAPI EGL interop
[libmpv_render/vaapi-egl] Trying to open a x11 VA display...
[libmpv_render/vaapi-egl] Trying to open a wayland VA display...
[libmpv_render/vaapi-egl] Trying to open a drm VA display...
[libmpv_render/vaapi-egl] Could not create a VA display.
[libmpv_render] Loading failed.
[libmpv_render] Loading hwdec driver 'cuda-nvdec'
Cannot load libcuda.so.1
[libmpv_render/cuda-nvdec] Failed to load CUDA symbols
[libmpv_render] Loading failed.
[libmpv_render] Loading hwdec driver 'drmprime-drm'
[libmpv_render/drmprime-drm] Failed to retrieve DRM fd from native display.
[libmpv_render] Loading failed.
hls executables located ->  {
  ffmpeg: '/usr/bin/ffmpeg',
  ffsplit: null,
  ffprobe: '/usr/bin/ffprobe'
}
Using app path -> /home/xtesting/.stremio-server
Enabling casting...
(node:34) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
EngineFS server started at http://127.0.0.1:11470
EngineFS HTTPS endpoint at https://local.strem.io:12470
-> GET /local-addon/manifest.json 
-> GET /settings 
➜  app 
```

**Comment by urisimchoni** [comment]
I just figured this thing out a couple of weeks ago, at least for my case - sorry for not coming back here to report. It wasn't a stremio issue at all, it had to do with the flatpak container and its permissions on the host.

To see this is the case I ran a shell on the flatpak - ```flatpak run --command=sh com.stremio.Stremio``` and saw that the shell would die on me after the same 10 seconds. So stremio itself is definitely not to blame.

With this new direction in mind I googled some more (not sure where's the article I've found) and found the fault is with the stremio flatpak not having permission to run in the background. AFAIR I fixed it on my system via the KDE flatpak settings UI, but now I don't see this permission anymore as a configurable thing.
